src/Server/Player/PlayerLogic.py

```diff
@@ -174,3 +185,6 @@ def _do_physics(self):
         elif self.pos.x > self.X_MAX:
             self.pos.x = self.X_MAX
             self._vel.x = -self._vel.x
+
+    def is_moving(self):
+        return abs(self._vel.x) > 0.001 and abs(self._vel.y) > 0.001
```

Comment:
Nope. It took me a while to find that this is the reason why I cannot attack... on horizontal ground.
Vector2D has mag getter, use that. Why should someone move BOTH in x and y direction?